File: backend/model/survey.go
package model

import (
	"github.com/jinzhu/gorm"
)

type Survey struct {
	gorm.Model
	Title        string `json:"title" gorm:"unique`
	Summary      string `json:"summary" gorm:"type:text"`
	Disclaimer   string `json:"disclaimer" gorm:"type:text"`
	Introduction string `json:"introduction"`
}

func (Survey) TableName() string { return "surveys" }

File: frontend/src/app/models/survey.ts
export class Survey {
  public ID: number;
  public CreatedAt: string;
  public UpdatedAt: string;
  public DeletedAt: string;
  public title: string;
  public summary: string;
  public disclaimer: string;
  public introduction: string;
}

export class Surveys {
  public surveys: Array<Survey>;
}

export class SurveyResponse {
  public survey: Survey;
}
